src/internal/clusterstate/migrationutils/pg_batcher.go

```diff
@@ -183,3 +182,63 @@ func (pb *postgresBatcher) execute(ctx context.Context) error {
 func (pb *postgresBatcher) Finish(ctx context.Context) error {
 	return pb.execute(ctx)
 }
+
+// SimplePostgresBatcher is an alternative batcher that buffers statements 'stmts' terminated by
+// semicolons until it hits the max number of statements. The statements are then executed in one
+// round-trip query to postgres.
+type SimplePostgresBatcher struct {
+	tx    *pachsql.Tx
+	stmts []string
+	max   int
+	num   int
+}
+
+func NewSimplePostgresBatcher(tx *pachsql.Tx) *SimplePostgresBatcher {
+	return &SimplePostgresBatcher{
+		tx:  tx,
+		max: 100,
+	}
+}
+
+// Add inserts a statement to the SimplePostgresBatcher's internal buffer. If the number
+// of statements exceeds the maximum after adding, the batcher executes the statements.
+// SimplePostgresBatcher doesn't support buffering arguments, the arguments are sanitized then
+// their values are inlined.
+//
+// Add does not support SqlNull<X> structs like SqlNullString.
+func (pb *SimplePostgresBatcher) Add(ctx context.Context, stmt string, args ...interface{}) error {
```

Comment:
At this point, it probably wouldn't be too misleading to call this `ExecContext`.